cpp/src/io/csv/csv_writer.cu

```diff
@@ -0,0 +1,258 @@
+ * @param[in] true_string String to use for 'true' values in boolean columns
+ * @param[in] false_string String to use for 'false' values in boolean columns
+ * @return NVStrings instance formated for CSV column output.
+ */
+NVStrings* column_to_strings_csv( gdf_column* column, const char* delimiter, const char* null_representation, const char* true_string, const char* false_string )
+{
+    NVStrings* rtn = nullptr;
+    gdf_size_type rows = column->size;
+    gdf_valid_type* valid = column->valid;
+    switch( column->dtype )
+    {
+        case GDF_STRING:
+            rtn = (static_cast<NVStrings*>(column->data))->copy();
+            break;
+        //case GDF_BOOL:
+        //    rtn = NVStrings::create_from_bools((const bool*)col->data,rows,true_string,false_string,valid);
+        //   break;
+        case GDF_INT32:
+            rtn = NVStrings::itos(static_cast<const int32_t*>(column->data),rows,valid);
+            break;
+        case GDF_INT64:
+            rtn = NVStrings::ltos(static_cast<const int64_t*>(column->data),rows,valid);
+            break;
+        case GDF_FLOAT32:
+            rtn = NVStrings::ftos(static_cast<const float*>(column->data),rows,valid);
+            break;
+        case GDF_FLOAT64:
+            rtn = NVStrings::dtos(static_cast<const double*>(column->data),rows,valid);
+            break;
+        case GDF_DATE64:
+            rtn = NVStrings::long2timestamp(static_cast<const uint64_t*>(column->data),rows,NVStrings::seconds,valid);
+            break;
+        default:
+            break;
+    }
+    CUDF_EXPECTS( rtn != nullptr, "write_csv: unsupported column type");
+
+    // replace nulls if specified
+    if( null_representation )
+    {
+        NVStrings* nstr = rtn->fillna(null_representation);
+        NVStrings::destroy(rtn);
+        rtn = nstr;
+    }
+
+    // probably could collapse this more
+    bool bquoted = (column->dtype==GDF_STRING);
+    // check for delimiters and quotes
+    bool* bmatches = nullptr;
+    RMM_TRY( RMM_ALLOC(&bmatches,rows*sizeof(bool),0) );
+    if( rtn->contains("\"",bmatches) > 0 )
+    {
+        NVStrings* esc = rtn->replace("\"","\"\"");
+        NVStrings::destroy(rtn);
+        rtn = esc;
+    }
+    else if( rtn->contains(",",bmatches) > 0 )
+        bquoted = true;
+    RMM_TRY( RMM_FREE( bmatches, 0 ) );
+    if( bquoted )
+    {
+        // prepend and append quotes if needed
+        NVStrings* pre = rtn->slice_replace("\"",0,0);
+        NVStrings::destroy(rtn);
+        rtn = pre->slice_replace("\"",-1,-1);
+        NVStrings::destroy(pre);
+    }
+    // append the delimiter last
+    if( delimiter && *delimiter )
+    {
+        NVStrings* dstr = rtn->slice_replace(delimiter,-1,-1);
+        NVStrings::destroy(rtn);
+        rtn = dstr;
+    }
+    return rtn;
+}
+
+/**---------------------------------------------------------------------------*
+ * @brief Creates CSV file from array of gdf_columns.
```

Comment:
Please don't duplicate doxygen document block. Doxygen should go in one place and one place only. For public APIs, that's the Public API header.

Reply:
Otherwise doxygen will raise errors when generating the HTML docs.

Reply:
Sorry, I don't understand. Delete this comment from the .cu file then?